### PR TITLE
Miscellaneous (mostly VM) improvements 20240403

### PR DESCRIPTION
**Pull request by RobertKrawitz**
- Retrieve qemu XML files from VMs
- Support multiple networks for VMs with multus (we have to do our own address assignment as CNV doesn't work with IPAM)
- Generate artifact directory names from timestamp to avoid inadvertently overwriting existing artifact directories.
- Retrieve all logs, not just from failing pods, in CI.
- Rename the stdout/stderr CI logs to reflect the change from kata to perf CI.
- Move the network interface options from uperf and server to top level to aid code reuse and simplify the logic.
- Change the reply from the sync to the workers to JSON so we can add more data to it if we need to.
- Ensure that all spreadsheet analysis is printed in numbers rather than suffixed human friendly.
- Files analysis should also analyze by direct vs. std IO.

### CODE REVIEW

**Comment by RobertKrawitz** [comment]
@gqlo @ebattat I'd mostly like you to look at the Multus-related changes for VMs and the retrieval of qemu XML files (both these are within clusterbuster)

File: CI/run-perf-ci-suite [hunk 1]
```diff
@@ -960,8 +967,8 @@ if ((! debugonly)) ; then
     fi
 
     if [[ -n "$artifactdir" && ! -d "$artifactdir" ]] ; then
-	exec > >(tee -a "$artifactdir/stdout.kata-perf-suite.log")
-	exec 2> >(tee -a "$artifactdir/stderr.kata-perf-suite.log" >&2)
+	exec > >(tee -a "$artifactdir/stdout.perf-ci-suite.log")
```
Comment (ebattat) [inline CI/run-perf-ci-suite:970]: Are we sure that removing the kata is necessary, or should we consider keeping it for future reference?

File: clusterbuster [hunk 2]
```diff
@@ -2132,6 +2201,19 @@ function _log_container() {
     fi
 }
 
+function _retrieve_vm_xml() {
+    local namespace=$1
+    local vm=$2
+    local clog="$artifactdir/VMXML/${namespace}_${vm}.xml"
+    if __OC exec -n "$namespace" "$(__OC get pod -n "$namespace" -l "vm.kubevirt.io/name=$vm" '-ojsonpath={.items[0].metadata.name}')" -- /bin/sh -c "cat '/var/run/libvirt/qemu/run/${namespace}_${vm}.xml'" > "${clog}.tmp" ; then
```
Comment (gqlo) [inline clusterbuster:2208]: By the way, `oc rsh -h` is getting more popular these days.

File: CI/run-perf-ci-suite (hunk 1 above)
Comment (RobertKrawitz) [reply CI/run-perf-ci-suite:970]: It's just renaming it, not removing it.

File: CI/run-perf-ci-suite [hunk 3]
```diff
@@ -650,6 +651,11 @@ function check_clusterbuster_option() {
     [[ -n "${known_clusterbuster_options[$option]:-}" ]]
 }
 
+function artifact_dirname() {
+    local basename=${1:-cb-ci-%T}
+    echo "${basename//%T/$(printf "%($snapshot_date_format)T" -1)}"
```
Comment (gqlo) [inline CI/run-perf-ci-suite:656]: I am curious what makes it different to do the following:
```
basename=${1:-cb-ci-T}
"${basename/%T/$(printf "%($snapshot_date_format)T" -1)}"
```

File: clusterbuster (hunk 2 above)
Comment (RobertKrawitz) [reply clusterbuster:2208]: I'm considering that.  At present it's succeeding, but the issue of `oc exec` not working in privileged pods is something I should think about.

File: CI/run-perf-ci-suite (hunk 3 above)
Comment (RobertKrawitz) [reply CI/run-perf-ci-suite:656]: If there are multiple occurrences of `%T` in the name they should all be expanded.

File: clusterbuster [hunk 4]
```diff
@@ -810,6 +842,11 @@ function inject_error() {
     warn "*** Registering error injection '$condition' = '${injected_errors[$condition]}'"
 }
 
+function artifact_dirname() {
+    local basename=${1:-cb-%T}
+    echo "${basename//%T/$(printf "%(%Y_%m_%dT%H_%M_%S%z)T" -1)}"
```
[the remark this answers is not in the document]
Comment (RobertKrawitz) [reply clusterbuster:847]: Yup, in progress.

**Review by gqlo — APPROVED** [review]
/LGTM

**Pull request merged**